runtime/near-vm-logic/src/gas_counter.rs

```diff
@@ -22,16 +22,19 @@ pub struct GasCounter {
     prepaid_gas: Gas,
     is_view: bool,
     ext_costs_config: ExtCostsConfig,
+    /// Where to store profile data, if needed.
+    profile: Option<&'a mut Vec<u64>>,
```

Comment:
Please hide it behind `costs_counting` flag and consider `Rc<RefCell<Vec<u64>>` also to avoid lifetime poisoning (when introduction of lifetime into one type leads to introduction of it in dependent types).